### DISCUSSION TITLE
Is there a profiler for Janet?

### DISCUSSION THREAD

**Discussion by MaxGyver83** [janet-lang/janet#1421] [Q&A] [answered]
If there is none, could this be possible as an external module or would this need to go into the Janet interpreter itself?

**Comment by ianthehenry** [comment]
Depending on the granularity you're looking for, `hyperfine` might work as a rough benchmarker. If you're trying to attribute cost centers to individual functions, I don't know of any existing way to do this. I think that you'd need probes inside the Janet vm on every function call?

You can profile individual functions yourself, but I don't think there's anything that will give you automatic attribution right now.

**Comment by sogaiu** [comment, marked as answer]
Somewhat along these lines, of interest might be:

* [janet-profiling](https://github.com/saikyun/janet-profiling)
* [spork/test's `timeit` and `timeit-loop`](https://github.com/janet-lang/spork/blob/86d7fd7e7dd06051ee044d47a566196c4170cc72/spork/test.janet#L61-L103)

**Comment by MaxGyver83** [comment]
`janet-profiling` is great! Thank you.

So sad that Google doesn't show me this project when I search for `janet profiler` or `janet profiling` or even `janet lang profiler` or `janet lang profiling`!
